gizmo2/src/main/java/org/jboss/jandex/gizmo2/Jandex2Gizmo.java

```diff
@@ -48,29 +50,33 @@
  * Bridge methods from {@code org.jboss.jandex} types to the Gizmo 2 API.
  */
 public class Jandex2Gizmo {
+    private static final Map<DotName, ClassDesc> CLASS_DESC_CACHE = new ConcurrentHashMap<>();
+
     /**
      * {@return the {@link ClassDesc} corresponding to the given Jandex {@link DotName}}
      * See {@link Type#name()} for the description of the format this method recognizes.
      *
      * @param name the Jandex {@code DotName} (must not be {@code null})
      */
     public static ClassDesc classDescOf(DotName name) {
-        if (name.prefix() == null) {
-            String local = name.local();
-            return switch (local) {
-                case "void" -> ConstantDescs.CD_void;
-                case "boolean" -> ConstantDescs.CD_boolean;
-                case "byte" -> ConstantDescs.CD_byte;
-                case "short" -> ConstantDescs.CD_short;
-                case "int" -> ConstantDescs.CD_int;
-                case "long" -> ConstantDescs.CD_long;
-                case "float" -> ConstantDescs.CD_float;
-                case "double" -> ConstantDescs.CD_double;
-                case "char" -> ConstantDescs.CD_char;
-                default -> ofClassOrArray(local);
-            };
-        }
-        return ofClassOrArray(name.toString());
+        return CLASS_DESC_CACHE.computeIfAbsent(name, nameParam -> {
```

Comment:
Probably worth using the same pattern as explained in: https://github.com/smallrye/jandex/pull/577#discussion_r2315516211 , especially since we parallelize class generation.